src/init.cpp

```diff
@@ -1177,7 +1178,15 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
 
     assert(!node.mempool);
     int check_ratio = std::min<int>(std::max<int>(args.GetArg("-checkmempool", chainparams.DefaultConsistencyChecks() ? 1 : 0), 0), 1000000);
-    node.mempool = std::make_unique<CTxMemPool>(node.fee_estimator.get(), check_ratio);
+
+    if (args.GetBoolArg("-mempool", DEFAULT_INITMEMPOOL)) {
+        LogPrintf("Mempool is initialized\n");
+        node.mempool = std::make_unique<CTxMemPool>(node.fee_estimator.get(), check_ratio);
+    }
+    else {
+        LogPrintf("Mempool is not initialized\n");
+        node.mempool = std::unique_ptr<CTxMemPool>(nullptr);
+    }
 
```

Comment:
```suggestion
    } else {
        LogPrintf("Mempool is not initialized\n");
    }

```

This is already a nullptr by default, so can skip this here

Reply:
Also, without a mempool the `nMempoolSizeMax` needs to be adjusted